auction-server/src/auction.rs

```diff
@@ -895,15 +908,48 @@ pub fn verify_submit_bid_instruction_svm(
                 .data
                 .starts_with(&express_relay_svm::instruction::SubmitBid::discriminator())
         })
-        .count()
-        != 1
-    {
-        return Err(RestError::BadParameters(
+        .cloned()
+        .collect();
+
+    match submit_bid_instructions.len() {
+        1 => Ok(submit_bid_instructions[0].clone()),
+        _ => Err(RestError::BadParameters(
             "Bid has to include exactly one submit_bid instruction to Express Relay program"
                 .to_string(),
-        ));
+        )),
     }
-    Ok(())
+}
+
+fn find_account_position_svm(idl: Idl, instruction: &str, account: &str) -> Option<usize> {
+    let instruction = idl.instructions.iter().find(|i| i.name == instruction)?;
+    instruction.accounts.iter().position(|a| match a {
+        IdlInstructionAccountItem::Single(a) => a.name == account,
+        IdlInstructionAccountItem::Composite(a) => a.name == account,
+    })
+}
+
+const SUBMIT_BID_INSTRUCTION_SVM: &str = "submit_bid";
+const PERMISSION_ACCOUNT_SVM: &str = "permission";
+
+fn extract_bid_data_svm(
+    idl: Idl,
+    accounts: &[Pubkey],
+    instruction: CompiledInstruction,
+) -> Result<(u64, Pubkey), RestError> {
```

Comment:
i think you also should extract the bid `deadline`? this could be relevant for deciding whether to include this bid in the auction or not (if deadline has already passed)

Reply:
We will run a simulation after the extraction. If the deadline is invalid, then the simulation will be failed.